core/cmd.go

```diff
@@ -63,27 +70,51 @@ func RunCmd(cwd string, args ...string) int {
 		close(sigChan)
 	}()
 
+	var exitCodeSet bool
+	var exitCode int
+	setExitCode := func(code int) {
```

Comment:
Is that necessary? If we receive from `waitCh`, we know the exit code for sure and can exit. If we receive on `timeoutCh`, we know that `waitCh` has not yet fired and thus set the code. If `waitCh` then fires, we do not override the code.

Thus, I would just return when receiving `timeoutCh` too

Reply:
I wanted to keep all logic related to waiting of exit of process (`waitCh`) and processing of the exit code in a single place so I went this way, not really sure what would be better